okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UserAgentInterceptor.java

```diff
@@ -28,14 +31,31 @@
     private static final Pattern VALID_USER_AGENT = Pattern.compile("[A-Za-z0-9()\\-#;/.,_\\s]+");
     private final String userAgent;
 
-    private UserAgentInterceptor(String userAgent) {
+    private UserAgentInterceptor(String userAgent, Class<?> serviceClass) {
         Preconditions.checkArgument(VALID_USER_AGENT.matcher(userAgent).matches(),
                 "User Agent must match pattern '%s': %s", VALID_USER_AGENT, userAgent);
-        this.userAgent = userAgent;
+
+        String version = getVersionString(serviceClass);
+        this.userAgent = version.isEmpty()
+                ? userAgent
+                : String.format("%s (%s)", userAgent, version);
+    }
+
+    private String getVersionString(Class<?> serviceClass) {
+        List<String> versions = new ArrayList<>(2);
+        String maybeRemotingVersion = this.getClass().getPackage().getImplementationVersion();
+        if (maybeRemotingVersion != null) {
+            versions.add("http-remoting " + maybeRemotingVersion);
+        }
+        String maybeServiceVersion = serviceClass.getPackage().getImplementationVersion();
+        if (maybeServiceVersion != null) {
+            versions.add(String.format("Service API %s %s", serviceClass.getSimpleName(), maybeServiceVersion));
```

Comment:
Would try to avoid String.format on this hot path as we could just do String concat and avoid the expensive format parsing

Reply:
Can do, but it's really not that hot: This executes once when making a JaxRsClient.

Reply:
done